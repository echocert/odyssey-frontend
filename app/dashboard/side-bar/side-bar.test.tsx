import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import SideBar from "./side-bar";
import { create } from "react-test-renderer";


describe("Side Bar", () => {
  it("Verify the snapshot of the SideBar section is matched", () => {
    const wrap = create(<SideBar />).toJSON();
    expect(wrap).toMatchSnapshot();
  });
  it("highlights the active item correctly", () => {
    const { getByTestId } = render(<SideBar activeIndex={0} />);
    const activeItem = getByTestId("side-bar-items").querySelector(".side-bar-item-active");
    expect(activeItem).not.toBeNull();
  });
  it('Checks the active state of the element', () => {
    const { getByTestId } = render(<SideBar activeIndex={0} />);
    const sideBarItems = getByTestId('side-bar-items');
    expect(sideBarItems.children[0]).toHaveClass('side-bar-item-active');
    expect(sideBarItems.children[1]).not.toHaveClass('side-bar-item-active');
    expect(sideBarItems.children[2]).not.toHaveClass('side-bar-item-active');
    // expect(sideBarItems.children[3]).not.toHaveClass('side-bar-item-active');
    // expect(sideBarItems.children[4]).not.toHaveClass('side-bar-item-active');
  });
});

