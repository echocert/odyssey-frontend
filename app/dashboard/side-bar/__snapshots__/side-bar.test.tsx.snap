// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Bar Verify the snapshot of the SideBar section is matched 1`] = `
<div
  className="side-bar"
  data-testid="side-bar"
>
  <aside
    className="aside-side-bar"
  >
    <div
      className="side-bar-logo-wrapper"
      onClick={[Function]}
    >
      <img
        alt="loading"
        className="side-bar-logo"
        src="./assets/images/odyssey_logo_large.png"
      />
    </div>
    <ul
      className="side-bar-item-container"
      data-testid="side-bar-items"
    >
      <li
        className="side-bar-item"
        data-testid="side-bar-properties"
        onClick={[Function]}
      >
        <div
          className="side-bar-item__contents"
        >
          <test-file-stub />
          <label
            className="label-hero-l2 label-header-l2 label-l2 label-header"
            style={
              {
                "marginLeft": "8px",
              }
            }
          >
            Properties
          </label>
        </div>
      </li>
       
      <li
        className="side-bar-item"
        onClick={[Function]}
      >
        <div
          className="side-bar-item__contents"
        >
          <div
            className="side-bar-item__notification"
          >
            <test-file-stub
              className="side-bar-item__icon"
            />
          </div>
          <label
            className="label-hero-l2 label-header-l2 label-l2 label-header"
            style={
              {
                "marginLeft": "8px",
              }
            }
          >
            Notifications
          </label>
        </div>
      </li>
      <li
        className="side-bar-item"
        onClick={[Function]}
      >
        <div
          className="side-bar-item__contents"
        >
          <div
            className="side-bar-item__notification"
          >
            <test-file-stub
              className="side-bar-item__icon"
            />
          </div>
          <label
            className="label-hero-l2 label-header-l2 label-l2 label-header"
            style={
              {
                "marginLeft": "8px",
              }
            }
          >
            Chat
          </label>
        </div>
      </li>
    </ul>
  </aside>
</div>
`;
